app/src/main/java/com/dsu/poc/repository/login/ValidationRepository.kt

```diff
@@ -0,0 +1,54 @@
+package com.dsu.poc.repository.login
+
+import android.app.Application
+import android.text.TextUtils
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+
+class ValidationRepository(application: Application) {
+
+    var application: Application
+
+    init {
+        this.application = application
+    }
+
+    fun validateCredentials(userName:String,password:String): LiveData<String> {
+        val loginErrorMessage = MutableLiveData<String>()
+        if(isUserNameValid(userName)){
+            if(!isPasswordValid(password)){
+                loginErrorMessage.value = "Invalid Password"
+            }else{
+                loginErrorMessage.value = "Successful Login"
+            }
+        }else{
+            loginErrorMessage.value = "Invalid Username"
+        }
+
+        return  loginErrorMessage
+    }
+
+
+
+
+
+    fun isUserNameValid(userName: String): Boolean {
+        if(!TextUtils.isEmpty(userName) && userName.equals("test"))
```

Comment:
replace .equals with "==" operator in Kotlin. Please keep addressing the IDE warnings